.travis.yml

```diff
@@ -23,28 +24,30 @@ env:
   ## Out of tree builds (default):
   # Debug build
   - BUILD_TYPE="Debug"
-  # Debug build (with SYMENGINE_ASSERT and Teuchos::RCP)
-  - BUILD_TYPE="Debug" WITH_SYMENGINE_ASSERT="yes" WITH_SYMENGINE_RCP="no"
```

Comment:
Why did you remove `WITH_SYMENGINE_RCP="no"` in all tests? We want to keep it, as this makes sure that all tests pass with the strict checking, thus ensuring the code is safe.